sdk/provisioning/Azure.Provisioning/src/operationalinsights/OperationalInsightsWorkspace.cs

```diff
@@ -0,0 +1,72 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using Azure.Core;
+using Azure.Provisioning.ResourceManager;
+using Azure.ResourceManager.OperationalInsights;
+using Azure.ResourceManager.OperationalInsights.Models;
+
+namespace Azure.Provisioning.OperationalInsights
+{
+    /// <summary>
+    /// Represents an Operational Insights workspace.
+    /// </summary>
+    public class OperationalInsightsWorkspace : Resource<OperationalInsightsWorkspaceData>
+    {
+        // https://learn.microsoft.com/azure/templates/microsoft.insights/2020-02-02/components?pivots=deployment-language-bicep
+        private const string ResourceTypeName = "Microsoft.Insights/components";
+        // https://github.com/Azure/azure-sdk-for-net/blob/main/sdk/operationalinsights/Azure.ResourceManager.OperationalInsights/src/Generated/RestOperations/WorkspacesRestOperations.cs#L36C42-L36C52
+        internal const string DefaultVersion = "2022-10-01";
+
+        private static readonly Func<string, OperationalInsightsWorkspaceData> Empty = (name) => ArmOperationalInsightsModelFactory.OperationalInsightsWorkspaceData();
```

Comment:
Because the private constructor has a Func parameter due to the fact that the public constructor needs to pass in a func so that the `name` can be provided in the base constructor.

Reply:
Methods implicitly convert to `Func`s and `Action`s if their signatures match.
```C#
using System;

static class App
{
    static string Empty(string x) => $"{x} is empty";
    static Func<string, string> Longer => x => $"{x} is still empty";

    static void Print(Func<string, string> formatter, string value) =>
        Console.WriteLine(formatter(value));

    public static void Main()
    {
        Print(Empty, "Foo");
        Print(Longer, "Foo");
    }
}
```

Reply:
Ah I misread your initial comment and didn't realize you were suggesting making it a method rather than just defining a static workspaceData Empty field.